packages/ckeditor5-typing/src/utils/inlinehighlight.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+import findAttributeRange from './findattributerange';
+
+/**
+ * @module typing/utils/inlinehighlight
+ */
+
+/**
+ * Adds a visual highlight style to an attribute element in which the selection is anchored.
+ * Together with two-step caret movement, they indicate that the user is typing inside the element.
+ *
+ * Highlight is turned on by adding the given class to the attribute element in the view:
+ *
+ * * The class is removed before the conversion has started, as callbacks added with the `'highest'` priority
+ * to {@link module:engine/conversion/downcastdispatcher~DowncastDispatcher} events.
+ * * The class is added in the view post fixer, after other changes in the model tree were converted to the view.
+ *
+ * This way, adding and removing the highlight does not interfere with conversion.
+ *
+ * @param {module:core/editor/editor~Editor} editor The editor instance.
+ * @param {String} attributeName The attribute name to check.
+ * @param {String} tagName The tagName of a view item.
+ * @param {String} className The class name to apply in the view.
+ */
+export default function setupLinkHighlight( editor, attributeName, tagName, className ) {
```

Comment:
Minimal sample need.

Reply:
And wrong function name.

Reply:
I think that the naming of the module/method must be improved. The default export should be named after the module AFAIR. Also it is used (in other PR) under totally different name and all of this is confusing.

Reply:
Another thing is mabye we should also expose this on `TwoStepCaretMovement` plugin?:

https://github.com/ckeditor/ckeditor5/pull/7577/files#diff-0efb47e00de0c1be680392e506f94f39R68-R71

One things is imported and the second is used from a plugin.

Reply:
> The default export should be named ...

The default export is named "default" by the definition :trollface: 

I change it all to `inlineHighlight`. I don't like it either :/ Do you have an idea for a better name? I got too attached to the former name, to figure out something new :|

Reply:
> Another thing is mabye we should also expose this on TwoStepCaretMovement plugin?:

I was considering it as well. Given how they were separated before, I thought it because of we foresee use-cases for 2SCM without any highlight.